### PR TITLE
improvise views, url management, add partial update support

### PR DESCRIPTION
**Pull request by jsparmani**
- Shifted to Model View Sets
- Add Partial Update Support
- Use DRF Routers for navigation

@Saurav-Shrivastav Kindly go through the changes once, I have tested them extensively on my side. Still check them once on ur side. Also, now are url patterns have changes

- For list view it's `/api/posts/`
- For create view it's `/api/posts/`
- For detail view it's `/api/posts/<pk>/`
- For update and partial update it's `/api/posts/<pk>/`
- For delete it's `/api/posts/<pk>/`
- For myposts it's `/api/posts/manage/myposts/`

### CODE REVIEW

File: buying_selling/posts/api/views.py [hunk 1]
```diff
@@ -23,39 +21,41 @@ def get_queryset(self, *args, **kwargs):
...
-    permission_classes = (IsAuthenticated,)
+
+    permission_classes_by_action = {
+        'create': [IsAuthenticated],
+        'list': [AllowAny],
+        'retrieve': [AllowAny],
+        'update': [IsAuthenticated, IsOwnerOrReadOnly],
+        'partial_update': [IsAuthenticated, IsOwnerOrReadOnly],
+        'destroy': [IsAuthenticated, IsOwnerOrReadOnly],
+    }
```
Comment (jsparmani) [inline buying_selling/posts/api/views.py:43]: @Saurav-Shrivastav Kindly verify the permission once to ensure I didn't make any mistakes

File: buying_selling/config/settings.py [hunk 2]
```diff
@@ -113,8 +113,8 @@
 SIGNING_KEY = env("SIGNING_KEY", default="",)
 
 SIMPLE_JWT = {
-    'ACCESS_TOKEN_LIFETIME': timedelta(minutes=5),
-    'REFRESH_TOKEN_LIFETIME': timedelta(days=1),
+    'ACCESS_TOKEN_LIFETIME': timedelta(days=1),
+    'REFRESH_TOKEN_LIFETIME': timedelta(days=2),
```
Comment (jsparmani) [inline buying_selling/config/settings.py:117]: Increased them for now, we will revoke these changes before production

Comment (Saurav-Shrivastav) [reply buying_selling/config/settings.py:117]: This was very much required, improves productivity

File: buying_selling/posts/api/views.py (hunk 1 above)
Comment (Saurav-Shrivastav) [reply buying_selling/posts/api/views.py:43]: Looks good to me.

**Pull request merged**